interfaces/dbus/backend.go

```diff
@@ -76,9 +76,9 @@ func (b *Backend) Deconfigure(snapInfo *snap.Info) error {
 
 // combineSnippets combines security snippets collected from all the interfaces
 // affecting a given snap into a content map applicable to EnsureDirState.
-func (b *Backend) combineSnippets(snapInfo *snap.Info, developerMode bool, snippets map[string][][]byte) (content map[string]*osutil.FileState, err error) {
+func (b *Backend) combineSnippets(snapInfo *snap.Info, snippets map[string][][]byte) (content map[string]*osutil.FileState, err error) {
 	for _, appInfo := range snapInfo.Apps {
-		if len(snippets) > 0 {
+		if len(snippets[appInfo.Name]) > 0 {
```

Comment:
That looks like a bug being fixed rather than a simplification. Can we have a test that has two apps, and one of them without any snippets, and roll that back to ensure the test is breaking before the problem is fixed?

Also, as a general hint, rather than indenting everything like this, this would read slightly more clear as:
```
for _, appInfo := range snapInfo.Apps {
        appSnippets := snippets[appInfo.Name]
        if len(appSnippets) == 0 {
                continue
        }
        ... more ...
}
```

Note how we can read that alone, even without knowing what "more" actually does.